### PR TITLE
Add MultiGetResponse::toString method

### PR DESCRIPTION
**Pull request by NeilRickards**
c.f. SearchResponse::toString

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java [hunk 1]
```diff
@@ -173,4 +173,17 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        try {
+            XContentBuilder builder = XContentFactory.jsonBuilder().prettyPrint();
+            builder.startObject();
+            toXContent(builder, EMPTY_PARAMS);
+            builder.endObject();
+            return builder.string();
+        } catch (IOException e) {
+            return "{ \"error\" : \"" + e.getMessage() + "\"}";
+        }
+    }
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java]: You can simply use `Strings.toString(this)` instead, at least that's what e.g. GetResponse does.

**Comment by cbuescher** [comment]
@NeilRickards thanks for this PR. I left a comment about how I think this can be made shorter. More generally I'm wondering if we should go down the road of adding `toString()` methods to all kinds of ActionResponses. There are a few that already do this, but a majority that don't. It might be useful to do this in a more generic way, I don't know if we ever had this discussion but maybe other can comment.

**Comment by NeilRickards** [comment]
That's much neater - I've updated the PR.
I wonder if there's scope for making `ActionResponse` implement `ToXContentObject` and overriding `toString` at that level.

**Comment by cbuescher** [comment]
From the looks of it, not all ActionResponses implement ToXContent/ToXContentObject so this will be difficult I think.

**Comment by javanna** [comment]
See #3889 . Making ActionResponse implement `ToXContentObject` should be possible, yet it may require quite some work. We could do it as part of adding support for new apis to the REST high level client.

As for adding `toString` in general, I even wonder if the `toString` should output the json representation. Anyways, I am ok with this change. Shall we have a test somewhere?

File: core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java (hunk 1 above)
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java]: Sorry, this doesn't compile now.

**Review by cbuescher — COMMENTED** [review]
@NeilRickards revisiting this after some time, whould you mind adding a short test for the toString() output like @javanna suggested?

**Comment by javanna** [comment]
I am closing this for now given that it's missing tests and we are not going to integrate it without them. Feel free to reopen once you have time to work on it. Thanks!

**Pull request closed without merging**